lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -387,19 +466,39 @@ class MetricsEndpointConsumer(Object):
 
     on = MonitoringEvents()
 
-    def __init__(self, charm, name):
+    def __init__(self, charm: CharmBase, relation_name: str = None):
         """A Prometheus based Monitoring service provider.
 
         Args:
             charm: a `CharmBase` instance that manages this
                 instance of the Prometheus service.
-            name: string name of the relation over which scrape target
+            relation_name: string name of the relation over which scrape target
                 information is gathered by the Prometheus charm.
         """
-        super().__init__(charm, name)
+        if not relation_name:
+            relation_interface = RELATION_INTERFACE_NAME
+            try:
+                # Check if there is just one relation with the right interface
+                relation_name = _get_single_relation_by_interface(
+                    charm, relation_interface, RelationDirection.REQUIRED
+                )
+            except NoRelationWithInterfaceFoundError:
```

Comment:
Well, in the beginning this PR did not have custom exception types. We *could* propagate the custom exception types, I have nothing against it.

Reply:
It felt also wrong to put the hints on how to fix it in the exception constructor or the `_get_single_relation_by_interface`, so I ended up catching the custom `Exception` type anyhow

Reply:
I would have thought propagating a custom exceptions is "safer" since then the Prometheus charm can completely define the semantics of these exceptions. I am not sure if the context in which we are raising a `ModelError` aligns with its intended semantics or if we are running the risk to overload `ModelError` too much. Will ping @jnsgruk for some deeper insight and a second opinion here.

Reply:
To me, this is fundamentally a "static" error (think of a compiler error, but for your charm), and not something the charm code should handle. My Python skills are not mighty enough to have an opinion about the best error/exception/curse to raise ;-)